accounts/README.md

```diff
@@ -1,10 +1,121 @@
 # Accounts
 
-The Safe{Core} Protocol is designed to be **account agnostic**. This initial alpha version sets a focus on the 1.x versions of Safe Smart Accounts to expedite the development process and gather feedback. These learnings are the foundation upon which the protocol is opened up to other account implementations.
+The Safe{Core} Protocol is designed to be **account agnostic**. This initial alpha version focuses on the 1.x versions of Safe Smart Accounts to expedite the development process and gather feedback. These learnings are the foundation upon which the protocol is opened to other account implementations.
 
-## Safe 1.4.0
+## High-level overview
 
-[Source Code](https://github.com/safe-global/safe-contracts/tree/v1.4.0)
+The [Safe{Core} Protocol Manager](https://github.com/safe-global/safe-core-protocol-specs/blob/2bffd759dd12be5583594f302d97c35e0ab9fcf5/manager/README.md) contract is the main entity interacting with accounts and vice versa. The Manager expects accounts to implement a specific interface, described in this document.
+
+## API Specification
+
+### General requirements
+
+The Account MUST append the 20 bytes of the `msg.sender` address to the calldata to each Account configuration-related method (setting a hook,
+adding a plugin, etc., for the full list, refer to [Manager's](../manager/README.md) specification). This is required because
+to support some functionality, such as function handlers, an Account may be required to execute a CALL
+operation that potentially bypasses the authorization to the Manager contract. This is required to allow the Manager to identify the transaction's sender. It is assumed
+that the only way to have the `msg.sender` equal to the Account address in the call frame is to execute a call through the Account's built-in authorization scheme.
+
+### Expected interface per Safe{Core} Protocol Module
+
+#### Plugins
+
+The account MUST implement the following interface:
+
+```Solidity
+interface IAccount {
+    function execTransactionFromModuleReturnData(
+        address to,
+        uint256 value,
+        bytes memory data,
+        uint8 operation
```

Comment:
@rmeissner - do you have context on the choice of using `uint8` instead of an `enum` for the operation?

I think that they are equivalent from an ABI standpoint, but an `enum` might be more self-documenting.